src/lib/emptyingSet.js

```diff
@@ -0,0 +1,24 @@
+ /**
+ * Copyright (c) 2016-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+module.exports = function emptyingSet() {
```

Comment:
I wanted to subclass, but babel transform + native Map don’t play nice. Thus the workaround. I will rename this to `<Something>Lock` and wrap the set (and expose `lock` and `unlock` methods).

Also: kill all the `new`s